src/openssl_stream.c

```diff
@@ -146,7 +151,10 @@ int git_openssl_stream_global_init(void)
 
 int git_openssl_set_locking(void)
 {
-#ifdef GIT_THREADS
+#ifndef GIT_THREADS
+	giterr_set(GITERR_THREAD, "libgit2 was not built with threads");
+	return -1;
```

Comment:
Just realized this does not make any sense when building against 1.1. I should probably just return success if `GIT_THREADS && OPENSSL_VERSION >= 0x10100000L`.